assert/utils_test.go

```diff
@@ -131,7 +131,7 @@ func ExampleHasPrefix() {
 func ExampleTrue() {
 	t := &T{}
 	assert.True(t, true)
-	assert.True(t, 1 == 1)
+	assert.True(t, 1 == 1) //nolint:staticcheck
```

Comment:
Is `// ` (rather than `//`) not allowed?

Reply:
The [`golangci-lint` documentation](https://golangci-lint.run/usage/false-positives/#nolint) claims:
> Use //nolint instead of // nolint because machine-readable comments should have no space by Go convention.

I don't know on what that claim is rooted.  However, there is this [related proposal](https://github.com/golang/go/issues/37974) about ignoring `//go:*` in doc comments.  The thing that was [ultimately merged](https://github.com/golang/go/commit/5a550b695117f07a4f2454039a4871250cd3ed09) ignores `//[a-z0-9]+:[a-z0-9]` (which notably doesn't handle the `//nolint` form suggested by the `golangci-lint` documentation).

Given all this, my answer is:

They are allowed, but it's nice to have them ignored in doc comments, so we probably want to elide the space there, and if we're eliding the space there, I think it makes sense for them to be consistent in non-doc comments, so I'm eliding the spaces here.

Reply:
Ah, good to know.

Reply:
I think the waters here are a little muddy. There has been some talk that Go should not take compiler directives with the same syntax as comments ([here](https://groups.google.com/g/golang-dev/c/r4rdPdsH1Fg/m/JG3Wlr0LwawJ?pli=1) and [here](https://news.ycombinator.com/item?id=9522973)), but Go has generally been OK saying that if the user does not have a space after `//`, then it is intended for the machine. While the Go compiler has followed that convention in {//go:*, //export, //line}, there are exceptions {// build, // import} and so other developers did not quite follow the convention and allow both with and without leading space.

Reply:
Looking now, we have all sorts of forms in our code:
```
// nolint: megacheck, structcheck, staticcheck
// nolint: errcheck
// nolint:errcheck
// nolint: structcheck,unused
//nolint:deadcode,unused
//nolint:gocyclo
//nolint:govet
```

(some generated by VDL).  It seems like the majority is `// nolint: errcheck`.  I think I never thought about it until this moment, so I'm probably personally responsible for some of the inconsistency 🤦 .